### ISSUE TITLE
model.n_classes assumed differently for binary classification.

### ISSUE THREAD

**Issue by mojganmadadi** [milesial/Pytorch-UNet#419]
Hi, 

As far as I saw, for binary classification the `model.n_classes` should be `2`, As input of the args parameters in [here](https://github.com/milesial/Pytorch-UNet/blob/master/train.py#L181) shows that. But throughout the code, in several places like [loss function initialization](https://github.com/milesial/Pytorch-UNet/blob/master/train.py#L81) and [loss calculation](https://github.com/milesial/Pytorch-UNet/blob/master/train.py#L102) it is asumed to be `1` for binary classification. Maybe I am wrong. Could you let me know what is the reason? Is it a bug or is there any reason behind it? thanks in advance.

**Comment by milesial** [comment]
Hi, it doesn't really matter if you use 1 or 2 classes for binary classification.
1 will use a single channel mask then sigmoid and threshold.
2 will use a 2-channels mask then argmax.
The losses for both are equivalent. CE with 2 classes is equivalent to BCE.

**Comment by gordon-n-stevenson** [comment]
Fixes a bug in the Dice loss that was n_classes not work 

https://github.com/milesial/Pytorch-UNet/pull/446
